src/main/java/com/google/cloud/tools/managedcloudsdk/install/TarGzExtractorProvider.java

```diff
@@ -53,7 +53,11 @@ public void extract(Path archive, Path destination, ProgressListener progressLis
     try (TarArchiveInputStream in = new TarArchiveInputStream(gzipIn)) {
       TarArchiveEntry entry;
       while ((entry = in.getNextTarEntry()) != null) {
-        final Path entryTarget = destination.resolve(entry.getName());
+        Path entryTarget = destination.resolve(entry.getName());
+
+        if (!Extractor.isTargetInsideDestination(destination, entryTarget)) {
+          throw new IOException("Blocked unzipping files outside destination: " + entry.getName());
```

Comment:
I wondered about using `java.nio.file.AccessDeniedException` rather than the very generic `IOException`.  Or `ZipException` and `ArchiveException`?

Reply:
I think I'll go for `ZipException` for both. `AccessDeniedException` seemed suitable actually, but its constructor expects the file in problem rather than a general error message.

Reply:
However, I won't expose `ZipException`, as this is an unusual case that shouldn't happen normally. `ZipException` is an `IOException`, so we don't need to change the method signature.

Reply:
`ZipException` doesn't seem right for the TarGz case.  I guess I'd keep `IOException`, or create a new specific exception — `InvalidArchiveEntryException`?  Your call.

Reply:
`IOException` doesn't seem totally out of place at the top level, generically stating the extraction operation failed. The TarGz API throws it if an archive is malformed. If I were to create `InvalidArchiveEntryException`, I'd make it a subclass of `IOException` as well, primarily to avoid exposing it to callers. This exceptional case is only about zips with malicious intent, which I don't want callers to be prepared for handling it in a special way.